src/components/recovery/RecoveryStatus/index.tsx

```diff
@@ -1,12 +1,34 @@
 import { SvgIcon, Typography } from '@mui/material'
+import { useContext } from 'react'
 import type { ReactElement } from 'react'
 
 import ClockIcon from '@/public/images/common/clock.svg'
 import { useRecoveryTxState } from '@/hooks/useRecoveryTxState'
 import type { RecoveryQueueItem } from '@/services/recovery/recovery-state'
+import { PendingStatus } from '@/store/pendingTxsSlice'
+import { RecoveryContext } from '../RecoveryContext'
+
+const STATUS_LABELS: Partial<Record<PendingStatus, string>> = {
+  [PendingStatus.SUBMITTING]: 'Submitting',
+  [PendingStatus.PROCESSING]: 'Processing',
+}
 
 export const RecoveryStatus = ({ recovery }: { recovery: RecoveryQueueItem }): ReactElement => {
   const { isExecutable, isExpired } = useRecoveryTxState(recovery)
+  const { pending } = useContext(RecoveryContext)
+
+  const status = pending?.[recovery.transactionHash] ? (
+    STATUS_LABELS[pending[recovery.transactionHash]]
```

Comment:
As discussed I think we need to use `recovery.args.txHash` here instead.

Reply:
The has was changed in c8a056f7ef0229bcfd4dc12036622ca21ef5ebbf and the spinner was added in b2fca640c6d6d7aea467fda126a3e0ffb97b26af.